### DISCUSSION TITLE
Proposal: online story editor

### DISCUSSION THREAD

**Discussion by EmplOTM** [endless-sky/endless-sky#10876] [Ideas]
Proposal: online story editor, like a map editor but for creating missions

Would it sound crazy to propose an online story editor that would automatically generate the script? 

As a long-time player fascinated by the story gameplay I've always wanted to participate and write missions

Creating stories with different outcomes is very demanding and I am sure most of the creatives who thought about contributing stopped at the amount of work it takes to learn how to put it into code

An app or a website that would allow to write missions and generate the code would be a big game changer
It could offer different areas ( am describing this just to help having a mental idea )
A page to storyboard the main steps and storyline into chapters
Each chapter could then be fine-tuned by filling text for the mission and choosing conditions from drop-down menus ( like the departure and arrival planet, the battle level for the mission to pop, recurring nature of the mission.. ) 
Then a bunch of files would be generated and downloaded in a zip file for local testing

Drop down menus for options and conditions would make things so much easier, would allow for a global overview of the story and surely bring in people that would love to enrich the game.
It is my personal belief that having more missions would greatly help balance the game. 

I imagine it would be a lot of work to make this real and my very small knowledge of programming makes me think it is possible.
 
Ideas anyone?

**Comment by Arteneko** [comment]
i'm starting to dig into trying to write side-stories so i'm gonna have to learn to implement these.
i'm mainly a web dev and have built similar things in the past so i'd probably give it a try at some point.
if i do i'll try to remember this discussion

**Reply by MaWe0208** [reply]
I just stumbled across this discussion and would like to know if there has been any tryouts so far. I would like to write some storys myself and like EmplOTM said my biggest problem is that I don't have any skills in coding.

**Reply by warp-core** [reply]
> I just stumbled across this discussion and would like to know if there has been any tryouts so far. I would like to write some storys myself and like EmplOTM said my biggest problem is that I don't have any skills in coding. 

I can't offer you an editor, but have you looked at some of the existing missions or the [creating missions wiki page](https://GitHub.com/endless-sky/endless-sky/wiki/CreatingMissions)? The game data doesn't really require traditional coding skills.

**Reply by MaWe0208** [reply]
> > I just stumbled across this discussion and would like to know if there has been any tryouts so far. I would like to write some storys myself and like EmplOTM said my biggest problem is that I don't have any skills in coding.
> 
> I can't offer you an editor, but have you looked at some of the existing missions or the [creating missions wiki page](https://GitHub.com/endless-sky/endless-sky/wiki/CreatingMissions)? The game data doesn't really require traditional coding skills.

Thanks for the information. The wiki you mentioned is somehow like the requested editor but without the drop down menu. 😅

**Comment by MaWe0208** [comment]
Now I just need to know where I can write these pseudocoding to make them become a real mission. 😅

**Reply by warp-core** [reply]
https://github.com/endless-sky/endless-sky/wiki/CreatingPlugins

**Reply by MaWe0208** [reply]
Thanks

**Comment by mustbeperfect** [comment]
Any updates on this? A visual editor would certainly be cool.

**Reply by CodeMagic** [reply]
I'm coming back to Endless Sky after a long while and hope to do some content creation this time around.  For anyone else interested, here are the existing tools I know about.  They're out of date and I haven't tried them yet, so no promises.  But, I only found them by digging through Discord and repos.  Now hopefully other folks don't have to do the same.

- https://github.com/quyykk/plugin-editor
- https://github.com/shitwolfymakes/Endless-Sky-Mission-Builder

I'm definitely interested in some kind of visual story editor, and have some ideas of how I would implement it; I'll come back to this discussion if I make any progress in that direction.

**Reply by MereDecade** [reply]
Not sure if it's useful, but I wrote [a simple C# data parser](https://gist.github.com/MereDecade/c65331609199bc1fe84b088907ea4497) that I use whenever I need to bulk-process ES data files. I'd be happy to port it into another language and/or extend it further if that'd be helpful for the mission editor.
